stdlib/public/core/Result.swift

```diff
@@ -0,0 +1,131 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A value that represents either a success or failure, capturing associated
+/// values in both cases.
+@_frozen
+public enum Result<Success, Failure: Error> {
```

Comment:
```suggestion
public enum Result<Success, Failure : Error> {
```
Space for conformamce or inheritance.

Reply:
We've recently stopped recommending that in the stdlib. We currently prefer not to use a space before the colon here.

Reply:
What's the reason for that? That was "the Apple way".

Reply:
I think it's best not to dwell too much on the pros and cons of indentation practices -- at the end of the day, these are random choices that mostly boil down to taste and personal background.

The closest thing we have to a "One True Swift Style" is what generated interfaces use. It is true that this current convention does not match that.

Reply:
(Deleted my previous comment that was some high-grade pretentious nonsense. Translation follows.)

There is no clever insight behind it: one day some of us working on the stdlib realized we like this style better, so we adopted it in new code. It’s the fashionable new stdlib thing for 2018, but I wouldn’t read too much into it.

Reply:
I don't blame anyone, I just want to understand the rationale behind this decision because I do follow the stdlib style in my own code base. See also the linked thread in this one [here](https://forums.swift.org/t/rfc-space-or-not-to-space/18657) on the forums.

Reply:
@lorentey I've removed all pre-colon spaces, take a look and see if it's what you expect.

Reply:
Looks good!